### PR TITLE
planner_cspace: add odom publisher to dummy robot

### PR DESCRIPTION
**Pull request by ykoga-kyutech**
Related to #59 
This PR add publisher for the odometry topic to use fake_localization for simulation.
fake_localization http://wiki.ros.org/fake_localization needs odometry topic (nav_msgs/Odometry)  to publish amcl_pose (geometry_msgs/PoseWithCovarianceStamped).

### CODE REVIEW

File: planner_cspace/src/dummy_robot.cpp [hunk 1]
```diff
@@ -70,30 +72,46 @@ class DummyRobot
...
 
+    pub_odom_ = nh.advertise<nav_msgs::Odometry>("odom", 1, true);
     sub_twist_ = nh.subscribe("cmd_vel", 1, &DummyRobot::cbTwist, this);
     sub_init_ = nh.subscribe("initialpose", 1, &DummyRobot::cbInit, this);
   }
   void spin()
   {
     const float dt = 0.01;
     ros::Rate rate(1.0 / dt);
+    ros::Time current_time = ros::Time::now();
```
Comment (at-wat) [inline planner_cspace/src/dummy_robot.cpp]: the value will never used

Comment (at-wat) [reply planner_cspace/src/dummy_robot.cpp]: this would be const in below `while` scope

File: planner_cspace/src/dummy_robot.cpp [hunk 2]
```diff
@@ -70,30 +72,46 @@ class DummyRobot
...
+      nav_msgs::Odometry odom;
+      odom.header.frame_id = "odom";
+      odom.header.stamp = current_time;
+      odom.header.seq++;
+      odom.child_frame_id = "base_link";
+      odom.pose.pose.position.x = x_;
+      odom.pose.pose.position.y = y_;
+      odom.pose.pose.orientation = tf::createQuaternionMsgFromYaw(yaw_);
+      odom.twist.twist.linear.x = v_;
+      odom.twist.twist.linear.y = 0.0;
```
Comment (at-wat) [inline planner_cspace/src/dummy_robot.cpp]: is it necessary?

File: planner_cspace/src/dummy_robot.cpp [hunk 3]
```diff
@@ -70,30 +72,46 @@ class DummyRobot
...
       trans.frame_id_ = "odom";
       trans.child_frame_id_ = "base_link";
       trans.setOrigin(tf::Vector3(x_, y_, 0.0));
       trans.setRotation(tf::createQuaternionFromYaw(yaw_));
       tfb_.sendTransform(trans);
+
+      nav_msgs::Odometry odom;
+      odom.header.frame_id = "odom";
+      odom.header.stamp = current_time;
+      odom.header.seq++;
```
Comment (at-wat) [inline planner_cspace/src/dummy_robot.cpp]: this doesn't effect.
incrementation of seq is done in ros client library.

**Comment by ykoga-kyutech** [comment]
@at-wat All done.

**Comment by at-wat** [comment]
BTW, dummy_robot is not install since it was for testing.
https://github.com/at-wat/neonavigation/blob/d41c22223fa3978c70047e8ce85f2e3cc17ba49d/planner_cspace/CMakeLists.txt#L76-L83
It seems better to make it installed before releasing this meta-package for rosdistro.

**Comment by ykoga-kyutech** [comment]
Would it be better to add the change in this PR?

**Comment by at-wat** [comment]
It would be ok to be contained in this PR.

**Pull request merged**